### PR TITLE
Reorganize directory structure

### PR DESCRIPTION
**Pull request by nshahan**
* Move all files out of lib/src/ directory. Files considered to be private to
    the package are within a src/ directory further down the hierarchy. For
    example: `lib/material_tab/src/`.

### CODE REVIEW

**Comment by nshahan** [comment]
I updated the branch on the examples to use the new locations of all files.
https://github.com/dart-lang/angular_components_example/tree/use_sass_mixins

**Comment by kevmoo** [comment]
See https://www.dartlang.org/tools/pub/package-layout#implementation-files

Would it be crazy to have all implementation files under lib/src ?

**Comment by nshahan** [comment]
If I am understanding your question correctly that would look like this:

```
├── lib
│   │ ...
│   ├── material_tab
│   │   └── material_tab.dart
│   │   │ ...
│   │ ...
│   ├── src
│   │   │ ...
│   │   └── material_tab
│   │   │   └── navigation_bar.dart
...
```

Yes that would be crazy. ;) That would make the internal google <--> github syncing story more complicated than I would like to maintain. It would also mean that the implementation for a component would then be in two places so you would have to bounce around the directory tree when trying to read or edit the source code.

**Comment by kevmoo** [comment]
@munificent – thoughts on this, oh arbiter of style?

**Comment by nshahan** [comment]
I had a bunch of offline conversations about the layout of these directories. I've decided it's not _that_ crazy. 

I'm evaluating how I can automate the syncing process better and hopefully reduce the burden when trying to add new files to the sync.

**Comment by nshahan** [comment]
I've updated this branch with all the private files moved to lib/src/. Ready to review changes again.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,12 +1,10 @@
 ## 0.8.0-dev
 
-  * Move all files out of lib/src/ directory. Files considered to be private to
-    the package are within a src/ directory further down the hierarchy. For
-    example: `lib/material_tab/src/`.
-  * Remove all precompiled .css files.
-  * Update all import statements in .scss files to use dart style package
-    imports.
-  * Add dependency on sass_builder package to compile .css files.
+ * Move entry points to all components all out of the lib/src/ directory.
```
Comment (kwalrath) [inline CHANGELOG.md]: all components all -> all components
?

Comment (nshahan) [reply CHANGELOG.md]: Yes! thanks.

**Comment by nshahan** [comment]
@kevmoo PTAL

**Pull request merged**